src/benchmarks/micro/libraries/System.Numerics.Tensors/Perf_SignedIntegerTensorPrimitives.cs

```diff
@@ -0,0 +1,128 @@
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.SIMD, Categories.JIT)]
+    [GenericTypeArguments(typeof(sbyte))]
+    [GenericTypeArguments(typeof(short))]
+    [GenericTypeArguments(typeof(int))]
+    [GenericTypeArguments(typeof(long))]
+    public class Perf_SignedIntegerTensorPrimitives<T>
+        where T : unmanaged, IBinaryInteger<T>, IMinMaxValue<T>
+    {
+        [Params(3, 64 + 7, 128 + 7, 256 + 7, 512 + 7, 3 * 512, 6 * 512)]
+        public int BufferLength;
+
+        private const int ShiftAmount = 3;
+
+        private T[] _source1;
+        private T[] _source2;
+        private T _scalar1;
+        private T[] _ones;
+        private T[] _destination;
+
+        [GlobalSetup]
+        public void Init()
+        {
+            _source1 = ValuesGenerator.Array<T>(BufferLength, seed: 42);
+            _source2 = ValuesGenerator.Array<T>(BufferLength, seed: 43);
+            _scalar1 = ValuesGenerator.Value<T>(seed: 44);
+            _ones = Enumerable.Repeat(T.One, BufferLength).ToArray();
+            _destination = new T[BufferLength];
+        }
+
+        #region Unary Operations
+        [Benchmark]
+        public void Abs() => TensorPrimitives.Abs<T>(_source1, _destination);
+
+        [Benchmark]
+        public void LeadingZeroCount() => TensorPrimitives.LeadingZeroCount<T>(_source1, _destination);
+
+        [Benchmark]
+        public void OnesComplement() => TensorPrimitives.OnesComplement<T>(_source1, _destination);
+
+        [Benchmark]
+        public void PopCount() => TensorPrimitives.PopCount<T>(_source1, _destination);
+
+        [Benchmark]
+        public void ShiftLeft() => TensorPrimitives.ShiftLeft<T>(_source1, ShiftAmount, _destination);
+
+        [Benchmark]
+        public void ShiftRightArithmetic() => TensorPrimitives.ShiftRightArithmetic<T>(_source1, ShiftAmount, _destination);
+
+        [Benchmark]
+        public void ShiftRightLogical() => TensorPrimitives.ShiftRightLogical<T>(_source1, ShiftAmount, _destination);
+
+        [Benchmark]
+        public void RotateLeft() => TensorPrimitives.RotateLeft<T>(_source1, ShiftAmount, _destination);
+        
+        [Benchmark]
+        public void RotateRight() => TensorPrimitives.RotateLeft<T>(_source1, ShiftAmount, _destination);
+
+        [Benchmark]
+        public void TrailingZeroCount() => TensorPrimitives.TrailingZeroCount<T>(_source1, _destination);
+        #endregion
+
+        #region Binary Operations
+        [Benchmark]
+        public void BitwiseAnd_Vector() => TensorPrimitives.BitwiseAnd<T>(_source1, _source2, _destination);
+
+        [Benchmark]
+        public void BitwiseAnd_Scalar() => TensorPrimitives.BitwiseAnd(_source1, _scalar1, _destination);
+
+        [Benchmark]
+        public void BitwiseOr_Vector() => TensorPrimitives.BitwiseOr<T>(_source1, _source2, _destination);
+
+        [Benchmark]
+        public void BitwiseOr_Scalar() => TensorPrimitives.BitwiseOr(_source1, _scalar1, _destination);
```

Comment:
I don't think we need both And and Or... they're effectively the same implementation, just with a & or |.